test/src/com/xilinx/rapidwright/edif/TestEDIFCell.java

```diff
@@ -117,4 +117,21 @@ public void testIsUniquifiedRemoveAndAdd() {
         Assertions.assertEquals(2, picoblazeTop.getNonHierInstantiationCount());
         Assertions.assertEquals(1, kcpsm6.getNonHierInstantiationCount());
     }
+
+    @Test
+    public void testRenamePort() {
+        Design design = RapidWrightDCP.loadDCP("bnn.dcp");
+        EDIFNetlist netlist = design.getNetlist();
+        EDIFCell topCell = netlist.getTopCell();
+
+        EDIFPort p = topCell.getPort("dmem_i_V_ce0");
+        topCell.renamePort("dmem_i_V_ce0", "test_port");
+
+        Assertions.assertNull(topCell.getPort("dmem_i_V_ce0"));
```

Comment:
Maybe consider a few other corner cases (if ports like this don't exist, you could always add them):
- `my_signal[5:2]`
- `my_signal[0:0]`
- `my_signal[5:5]`